File: 53-maximum-subarray/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {
    let maxi = -Infinity
    let sum = 0

    for (let i = 0; i < nums.length; i++) {
        if(nums.length == 0){
            return nums[i]
        }
        sum = sum + nums[i]

        if(sum > maxi){
            maxi = sum
        }

        if(sum < 0 ){
            sum = 0
        }
    }

    return maxi
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
  public int maxSubArray(int[] nums) {
    int max = Integer.MIN_VALUE;
    int sum = 0;

    if (nums.length == 0) {
      return nums[0];
    }

    for (int num : nums) {
      // if (nums.length == 0) {
      //   return nums[i];
      // }
      sum += num;

      max = Math.max(sum, max);
      if (sum < 0) {
        sum = 0;
      }

    }

    return max;
  }
}